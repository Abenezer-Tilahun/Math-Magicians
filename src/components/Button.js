import React from 'react';
import PropTypes from 'prop-types';

export default class Button extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { value, onClick } = this.props;
    let className = /[0-9.AC/%/]/.test(value) ? 'btn' : 'btn btn-operation';
    className += value === 'AC/%/+/-' ? 'btn' : '';
    className += value === '0' ? ' btn-0' : '';

    return (
      <button
        className={className}
        onClick={() => onClick(value)}
        type="button"
      >
        {value}
      </button>
    );
  }
}

Button.propTypes = {
  value: PropTypes.string.isRequired,
  onClick: PropTypes.func,
};

Button.defaultProps = {
  onClick: () => null,
};
