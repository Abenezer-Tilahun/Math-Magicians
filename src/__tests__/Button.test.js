import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../components/Button';

it('renders correctly', () => {
  const tree = renderer
    .create(<Button value="+" onClick={() => { }} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
