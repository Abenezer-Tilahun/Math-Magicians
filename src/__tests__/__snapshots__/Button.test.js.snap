// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="btn btn-operation"
  onClick={[Function]}
  type="button"
>
  +
</button>
`;
